test/data_base.cpp

```diff
@@ -63,150 +116,45 @@ static void test_block_exists(const data_base& interface, size_t height,
-            //     auto history = address_store.get(address.hash());
-            //     auto found = false;
-               
-            //     for (const payment_record& row: history)
-            //     {
-            //         BOOST_REQUIRE(row.is_valid());
-
-            //         if (row.hash() == tx_hash && row.index() == j)
-            //         {
-            //             BOOST_REQUIRE_EQUAL(row.height(), height);
-            //             BOOST_REQUIRE_EQUAL(row.data(), output.value());
-            //             found = true;
-            //             break;
-            //         }
-            //     }
-
-            //     BOOST_REQUIRE(found);
-            // }
+            test_inputs_cataloged(payments_store, tx, true);
         }
+
+        test_outputs_cataloged(payments_store, tx, true);
+        ++tx_position;
     }
 }
 
 static void test_block_not_exists(const data_base& interface,
-    const block& block0, bool index_addresses)
+    const block& block, bool catalog)
 {
-    const auto& address_store = interface.addresses();
+    const auto& payments_store = interface.addresses();
 
     // Popped blocks still exist in the block hash table, but not confirmed.
-    const auto block_hash = block0.hash();
+    const auto block_hash = block.hash();
     const auto result = interface.blocks().get(block_hash);
     BOOST_REQUIRE(!is_confirmed(result.state()));
 
-    for (size_t i = 0; i < block0.transactions().size(); ++i)
+    uint32_t tx_position = 0u;
+
+    for (const auto& tx: block.transactions())
     {
-        const auto& tx = block0.transactions()[i];
-        const auto tx_hash = tx.hash();
+        if (!catalog)
+            return;
 
         if (!tx.is_coinbase())
         {
-            for (size_t j = 0; j < tx.inputs().size(); ++j)
-            {
-                const auto& input = tx.inputs()[j];
-                // auto r0_spend = interface.spends().get(input.previous_output());
-                // BOOST_REQUIRE(!r0_spend.is_valid());
-
-                if (!index_addresses)
-                    continue;
-
-                const auto addresses = input.addresses();
-                ////const auto& prevout = input.previous_output();
-                ////const auto address = prevout.metadata.cache.addresses();
-
-                // for (const auto& address: addresses)
-                // {
-                //     auto history = address_store.get(address.hash());
-                //     auto found = false;
-
-                //     for (const payment_record& row: history)
-                //     {
-                //         if (row.hash() == tx_hash && row.index() == j)
-                //         {
-                //             found = true;
-                //             break;
-                //         }
-                //     }
-
-                //     BOOST_REQUIRE(!found);
-                // }
-            }
+            test_inputs_cataloged(payments_store, tx, true);
```

Comment:
Shouldn’t catalog (vs true) be passed here (and below)?

Reply:
I took out the `catalog` parameter, as I want these helpers to be called only if cataloging is enabled, so the `catalog` parameter is redundant.

The true|false instead captures if the payment should be found in the address database (aka payment store here). Recall I renamed a badly named argument from `is_found` to `expect_found`. This true|false is for that argument.